### ISSUE TITLE
`ipfs files` (MFS) mixes up files and directories on `ipfs files mv`

### ISSUE THREAD

**Issue by alexander255** [ipfs/go-ipfs#2654]
Try running this:

```sh
ipfs files mkdir /test_dir
echo "content" | ipfs files write -e "/test_file"
ipfs files rm -r "/test_dir"
ipfs files mv "/test_file" "/test_dir"
ipfs files stat "/test_dir"
ipfs files read "/test_dir"
```

# Expected result

```
QmZh2DtJpG7XxGaLrDCxMFFNz5ASgJnuy3LL4yAaumRND3
Size: 11
CumulativeSize: 69
ChildBlocks: 1
Type: file

content
```

# Actual result

```
QmZh2DtJpG7XxGaLrDCxMFFNz5ASgJnuy3LL4yAaumRND3
Size: 11
CumulativeSize: 69
ChildBlocks: 1
Type: directory

Error: /test_dir was not a file.
```

# Additional Notes

Doing `ipfs cat` on the hash produced by `ipfs files stat` works:

```sh
$ ipfs cat /ipfs/QmZh2DtJpG7XxGaLrDCxMFFNz5ASgJnuy3LL4yAaumRND3
content
lt
```

**Comment by whyrusleeping** [comment]
Thanks for reporting! I'll get this fixed ASAP

**Comment by whyrusleeping** [comment]
this was addressed in my mfs caching fix #2795
